src/App.tsx

```diff
@@ -1,32 +1,94 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
 import './App.css'
+import Navbar from './components/Navbar'
+import Categories from './components/Categories'
+import ProductPage from './components/ProductPage'
+import Cart from './components/Cart'
+import Route from './components/Route'
+import { useState, useMemo } from 'react'
+import ProductList from './components/ProductList'
+
+interface ProductCartItem {
+  id: string
+  image: string
+  title: string
+  price: number
+  category: string
+  description: string
+  quantity: number
+}
 
 function App() {
-  const [count, setCount] = useState(0)
+  // state variables
+  const [isCartOpen, setIsCartOpen] = useState(false)
+  const [cartProducts, setCartProducts] = useState<ProductCartItem[]>([])
+  const cartCount = useMemo(() => {
+    return cartProducts.reduce((count, product) => count + product.quantity, 0)
+  }, [cartProducts])
+
+  // cart handling
+  const toggleCart = () => setIsCartOpen(!isCartOpen)
+
+  const handleAddToCart = (newProduct: ProductCartItem) => {
+    // setCartCount(cartCount + 1); should no longer need due to memoization
```

Comment:
Yea this seems to work quite well! Can remove this commented code at this point imho